neps/nep-0508.md

```diff
@@ -0,0 +1,151 @@
+---
+NEP: 508
+Title: Resharding phase 2
+Authors: Waclaw Banasik, Shreyan Gupta, Yoon Hong
+Status: Draft
+DiscussionsTo: https://github.com/near/nearcore/issues/8992
+Type: Protocol
+Version: 1.0.0
+Created: 2022-09-19
+LastUpdated: 2023-09-19
+---
+
+## Summary
+
+In essence, this NEP is extension of [NEP-40](https://github.com/near/NEPs/blob/master/specs/Proposals/0040-split-states.md), which focused splitting one shard into multiple shards. 
+
+We are introducing the second phase of resharding, which supports one shard splitting into two within one epoch at pre-determined split boundary. 
+
+While the new approach addresses critical limitations left unsolved in NEP-40 and is expected to remain valid for foreseable future, it does not serve all usecases, such as dynamic resharding.
+
+## Motivation
+
+Currently, NEAR protocol has four shards. With more partners onboarding, we started seeing that some shards occasionally become over-crowded. In addition, with state sync and stateless validation, validators do not need to track all shards and validator hardware requirements can be greatly reduced with smaller shard size. 
+
+## Specification
+
+### High level assumptions
+
+* Some form of State sync (centralized or decentralized) is enabled.
+* Flat state is enabled.
+* Shard split boundary is predetermined. In other words, necessity of shard splitting is manually decided.
+* Merkle Patricia Trie is undelying data structure for the protocol.
+* Minimal epoch gap between two resharding events is X.
+
+### High level requirements
+
+* Resharding should work even when validators stop tracking all shards.
+* Resharding should work after stateless validation is enabled.
+* Resharding should be fast enough so that both state sync and resharding can happen within one epoch.
+* Resharding should not require additional hardware from nodes.
```

Comment:
I believe this is false. There are a few different reasons for increased hardware requirements.
- while resharding we need to store duplicate state for the shard that is being split
- increasing the number of shards naturally increases the total gas per block
- resharding and catchup are resource consuming

Reply:
I had doubt and am happy that we are having this conversation. If we need additional hardware specs, we need to specify what they should be. I don't think we have a good answer right now, but let's gather the data while testing.

Reply:
Agreed. I hope the node team can help us benchmark the additional hardware requirements as part of pre-release mocknet testing.